app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/fragments/DatasetListRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,73 @@
+package com.github.HumanLearning2021.HumanLearningApp.view.fragments
+
+import android.app.Activity
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.lifecycle.LiveData
+import androidx.recyclerview.widget.RecyclerView
+import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.model.Dataset
+
+class DatasetListRecyclerViewAdapter(
+    private val hostActivity: Activity,
+    private val datasetList: LiveData<List<Dataset>>,
+    private val itemClickedCallback: (Dataset) -> Unit
+) : RecyclerView.Adapter<DatasetListRecyclerViewAdapter.ListItemViewHolder>() {
+
+    /**
+     * Defines the number of categories shown on one ListItemViewHolder
+     */
+    private val NB_REPRESENTATIVES_SHOWN = 3
+
+    /**
+     * Gets called when a new view holder is needed
+     * (for example when the user scrolls to reveal more items)
+     */
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListItemViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.fragment_dataset_list_elt, parent, false)
+        return ListItemViewHolder(view)
+    }
+
+    /**
+     * Gets called to allow the ViewHolder to be bound to its data, given its position
+     */
+    override fun onBindViewHolder(holder: ListItemViewHolder, position: Int) {
+        val ds = datasetList.value?.get(position)
+        holder.itemView.apply {
+            if (ds != null) {
+                findViewById<TextView>(R.id.DatasetList_elt_text).text = ds.name
+                val catList = ds.categories.toList()
+                Log.d("onBindViewHolder", catList.toString())
+                if (catList.size >= NB_REPRESENTATIVES_SHOWN) {
+                    catList[0].representativePicture?.displayOn(
+                        hostActivity,
+                        findViewById(R.id.DatasetList_elt_im0)
+                    )
+                    catList[1].representativePicture?.displayOn(
+                        hostActivity,
+                        findViewById(R.id.DatasetList_elt_im1)
+                    )
+                    catList[2].representativePicture?.displayOn(
+                        hostActivity,
+                        findViewById(R.id.DatasetList_elt_im2)
+                    )
```

Comment:
We will need to have a discussion about the representativePictures. In the DummyDS, no category have a repr. picture which, for me, is a problem. Imo, it would be nice if we "force" to have a representative picture for each category. My idea was : When you add a category to your DS, it launches the camera activity to take a picture which will be the representative of the category, and if a representative picture is deleted from a DS, the first picture we find that belongs in the same category becomes the new representative picture for example. Anyway I think it would be nice to be certain that each category have a representative picture at any point of time to avoid unecessary issues, but we should have a discussion about this I think?

Reply:
Yes I agree, it's a bit of a problem to have categories without representatives. Let's discuss this tomorrow maybe